include/kamping/p2p/sendrecv.hpp

```diff
@@ -0,0 +1,267 @@
+// This file is part of KaMPIng.
+//
+// Copyright 2022-2024 The KaMPIng Authors
+//
+// KaMPIng is free software : you can redistribute it and/or modify it under the
+// terms of the GNU Lesser General Public License as published by the Free
+// Software Foundation, either version 3 of the License, or (at your option) any
+// later version. KaMPIng is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+// for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with KaMPIng.  If not, see <https://www.gnu.org/licenses/>.
+
+#pragma once
+
+#include <type_traits>
+#include <utility>
+
+#include <kassert/kassert.hpp>
+#include <mpi.h>
+
+#include "kamping/communicator.hpp"
+#include "kamping/data_buffer.hpp"
+#include "kamping/implementation_helpers.hpp"
+#include "kamping/named_parameter_check.hpp"
+#include "kamping/named_parameter_selection.hpp"
+#include "kamping/named_parameter_types.hpp"
+#include "kamping/named_parameters.hpp"
+#include "kamping/p2p/helpers.hpp"
+#include "kamping/p2p/probe.hpp"
+#include "kamping/parameter_objects.hpp"
+#include "kamping/result.hpp"
+#include "kamping/status.hpp"
+
+///// @addtogroup kamping_p2p
+/// @{
+
+// @brief Wrapper for \c MPI_Sendrecv.
+///
+/// This wraps \c MPI_Sendrecv. This operation performs a blocking send and receive operation. If the
+/// \ref kamping::recv_counts() parameter is not specified, this first performs a probe, followed by a receive of
+/// the probed message with the probed message size.
+///
+/// The following parameters are required:
+/// - \ref kamping::send_buf() containing the data that is sent.
+///
+///
+/// - \ref kamping::destination() the receiving rank.
+///
+/// The following parameter is optional, but leads to an additional call to \c MPI_Probe if not present:
```

Comment:
```suggestion
/// The following parameter is optional, but leads to an additional call to \c MPI_Sendrecv if not present:
```

Reply:
Also clarify, that the receive count must either be provided on both ranks taking part in the communication, or on none. (otherwise the code will block)